modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -280,6 +279,15 @@ CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
   // No subgenerators will be called if option to bypass mesh generators is enabled
   if (!getReactorParam<bool>(RGMB::bypass_meshgen))
   {
+    // Check whether flexible stitching should be used for constituent assemblies and throw a
+    // warning if flexible stitching option is not enabled
+    if (!getReactorParam<bool>(RGMB::flexible_assembly_stitching) &&
+        constituentAssembliesNeedFlexibleStiching())
+      mooseWarning("Constituent assemblies do not share the same number of nodes at the outer "
+                   "boundary. In order to ensure that output mesh does not having hanging nodes, a "
+                   "flexible stitching approach should be used by setting "
+                   "ReactorMeshParams/flexible_assembly_stitching = true.");
```

Comment:
do we have a location we can provide? or some sort of Id of the assemblies? the user will want to see those hanging nodes

Reply:
Before this warning, there will be an additional warning message that prints out the assembly names where dissimilarity is detected